site/tests/app/libraries/LoggerTester.php

```diff
@@ -104,8 +103,34 @@ public function testInvalidDirectory() {
         $this->assertEquals($log_path, Logger::getLogPath());
     }
 
+    public function testTALog(){
+        $_SERVER['REMOTE_ADDR'] = "127.0.0.1";
+        $_SERVER['HTTP_USER_AGENT'] = "PHPUnit";
+        $logging_params = array("gradeable_id" => "test_gradeable",
+                                "grader_id" => "test_grader",
+                                "submitter_id" => "test_submitter",
+                                "action" => "test_action",
+                                "component_id" => "1");
+        Logger::logTAGrading($logging_params);
+        $file = file_get_contents($this->ta_grading);
+        $lines = explode("\n", $file);
+        $this->assertCount(2, $lines);
+        $this->assertEmpty($lines[1]);
+        $line = explode(" | ", $lines[0]);
+        $datetime = explode(" ", $line[0]);
+        $time = explode(":", $datetime[0]);
+        $date = explode("/", $datetime[1]);
+        $this->assertTimeEqualsCurrent($date, $time);
+        $this->assertEquals(2, strlen($time[2]));
+        $this->assertEquals("test_gradeable", $line[1]);
+        $this->assertEquals("instructor", $line[2]);
+        $this->assertEquals("aphacker", $line[3]);
+        $this->assertEquals("1", $line[4]);
+        $this->assertEquals("VIEW_PAGE", $line[5]);
```

Comment:
It looks like these asserts should be (based on the `$logging_params`):
```
        $this->assertEquals("test_grader", $line[2]);
        $this->assertEquals("test_submitter", $line[3]);
        $this->assertEquals("1", $line[4]);
        $this->assertEquals("test_action", $line[5]);
```

Reply:
Oh my goodness I keep changing one line of code each time thinking that was the only mistake. Sorry this took so many tries